ark/utils/segmentation_utils.py

```diff
@@ -176,3 +176,39 @@ def concatenate_csv(base_dir, csv_files, column_name="point", column_values=None
             combined_data = pd.concat((combined_data, temp_data), axis=0, ignore_index=True)
 
     combined_data.to_csv(os.path.join(base_dir, "combined_data.csv"), index=False)
+
+
+def visualize_segmentation(segmentation_labels_xr, channel_data_xr,
+                           output_dir, chan_list=None, fovs=None):
+    """For each fov, generates segmentation labels, segmentation borders, and overlays
+    over the channels in chan_list if chan_list is provided.
+    Saves xarray to output directory
+
+    Args:
+        segmentation_labels_xr (xarray.DataArray): xarray containing segmentation labels
+        channel_data_xr (xarray.DataArray): xarray containing TIFs
+        output_dir (str): path to directory where the output will be saved
+        chan_list (list): list of channels to overlay segmentation output over
+        fovs (numpy.ndarray): field of view
+    """
+
+    if fovs is None:
+        fovs = segmentation_labels_xr.fovs
+    for fov in fovs:
+        labels = segmentation_labels_xr.loc[fov, :, :, 'whole_cell'].values
+        # If chan_list is provided, overlay segmentation output over it
+        if chan_list is not None:
+            input_data = channel_data_xr.loc[fov, :, :, chan_list].values
+            save_path = '_'.join([f'{fov}', *chan_list.astype('str'), 'overlay.tiff'])
+            plot_utils.plot_overlay(
+                labels,
+                plotting_tif=input_data,
+                path=os.path.join(output_dir, save_path)
+            )
+
+        plot_utils.plot_overlay(
+            labels,
+            plotting_tif=None,
+            path=os.path.join(output_dir, f'{fov}_segmentation_borders.tiff')
+        )
+        io.imsave(os.path.join(output_dir, f'{fov}_segmentation_labels.tiff'), labels)
```

Comment:
Now that I think about it, do we need either of these @ngreenwald?  This is just saving the borders and then the labels separately, but I don't think we need the out-of-context borders, and the tiff labels would just be the deepcell output.  I think these lines can be deleted, and the test functions updated accordingly.

Reply:
The out of context borders are useful for creating photoshop overlays. The tiff labels may end up being the same as the deepcell_output; depends on how Omer's PR finishes up. We can leave for now and remove the pure labels if he ends up implementing same thing